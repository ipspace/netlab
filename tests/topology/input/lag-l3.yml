#
# Basic L3 LAG example - single lag with 2 member links
#

defaults:
  provider: clab
  device: dellos10

module: [lag, gateway]
nodes: [r1, r2]
links:
- lag.members: [r1-r2, r1-r2]

- prefix: 10.11.12.0/24        # Custom link prefix
  lag.members:
  - r1:
      gateway.protocol: vrrp   # Gateway enabled on interface
      ipv4: 12                 # Custom relative IP
    r2:
  - r1:
    r2:
