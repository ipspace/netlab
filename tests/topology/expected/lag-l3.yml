input:
- topology/input/lag-l3.yml
- package:topology-defaults.yml
lag:
  lacp: fast
  lacp_mode: active
links:
- _linkname: links[1]
  bridge: input_1
  interfaces:
  - ifindex: 30000
    ifname: port-channel1
    ipv4: 10.1.0.1/30
    lag:
      ifindex: 1
    node: r1
  - ifindex: 30000
    ifname: port-channel1
    ipv4: 10.1.0.2/30
    lag:
      ifindex: 1
    node: r2
  lag: {}
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
  type: lag
- _linkname: links[1].lag[1]
  interfaces:
  - ifindex: 1
    ifname: ethernet1/1/1
    node: r1
  - ifindex: 1
    ifname: ethernet1/1/1
    node: r2
  lag:
    _parentindex: 1
  linkindex: 2
  node_count: 2
  prefix: false
  type: p2p
- _linkname: links[1].lag[2]
  interfaces:
  - ifindex: 2
    ifname: ethernet1/1/2
    node: r1
  - ifindex: 2
    ifname: ethernet1/1/2
    node: r2
  lag:
    _parentindex: 1
  linkindex: 3
  node_count: 2
  prefix: false
  type: p2p
module:
- lag
- vlan
name: input
nodes:
  r1:
    af:
      ipv4: true
    box: vrnetlab/vr-ftosv
    clab:
      kind: vr-ftosv
    device: dellos10
    hostname: clab-input-r1
    id: 1
    interfaces:
    - ifindex: 30000
      ifname: port-channel1
      ipv4: 10.1.0.1/30
      lag:
        ifindex: 1
        lacp: fast
        lacp_mode: active
      linkindex: 1
      mtu: 1500
      name: r1 -> r2
      neighbors:
      - ifname: port-channel1
        ipv4: 10.1.0.2/30
        node: r2
      type: lag
      virtual_interface: true
    - clab:
        name: eth1
      ifindex: 1
      ifname: ethernet1/1/1
      lag:
        _parentindex: 1
      linkindex: 2
      mtu: 1500
      name: r1 -> r2
      neighbors:
      - ifname: ethernet1/1/1
        node: r2
      type: p2p
    - clab:
        name: eth2
      ifindex: 2
      ifname: ethernet1/1/2
      lag:
        _parentindex: 1
      linkindex: 3
      mtu: 1500
      name: r1 -> r2
      neighbors:
      - ifname: ethernet1/1/2
        node: r2
      type: p2p
    lag:
      lacp: fast
      lacp_mode: active
    loopback:
      ifindex: 0
      ifname: loopback0
      ipv4: 10.0.0.1/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: mgmt1/1/1
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    module:
    - lag
    - vlan
    mtu: 1500
    name: r1
  r2:
    af:
      ipv4: true
    box: vrnetlab/vr-ftosv
    clab:
      kind: vr-ftosv
    device: dellos10
    hostname: clab-input-r2
    id: 2
    interfaces:
    - ifindex: 30000
      ifname: port-channel1
      ipv4: 10.1.0.2/30
      lag:
        ifindex: 1
        lacp: fast
        lacp_mode: active
      linkindex: 1
      mtu: 1500
      name: r2 -> r1
      neighbors:
      - ifname: port-channel1
        ipv4: 10.1.0.1/30
        node: r1
      type: lag
      virtual_interface: true
    - clab:
        name: eth1
      ifindex: 1
      ifname: ethernet1/1/1
      lag:
        _parentindex: 1
      linkindex: 2
      mtu: 1500
      name: r2 -> r1
      neighbors:
      - ifname: ethernet1/1/1
        node: r1
      type: p2p
    - clab:
        name: eth2
      ifindex: 2
      ifname: ethernet1/1/2
      lag:
        _parentindex: 1
      linkindex: 3
      mtu: 1500
      name: r2 -> r1
      neighbors:
      - ifname: ethernet1/1/2
        node: r1
      type: p2p
    lag:
      lacp: fast
      lacp_mode: active
    loopback:
      ifindex: 0
      ifname: loopback0
      ipv4: 10.0.0.2/32
      neighbors: []
      type: loopback
      virtual_interface: true
    mgmt:
      ifname: mgmt1/1/1
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    module:
    - lag
    - vlan
    mtu: 1500
    name: r2
provider: clab
